### ISSUE TITLE
GRPO Training Generates Identical Completions for Same Prompts in Async Mode

### ISSUE THREAD

**Issue by zliu269** [modelscope/ms-swift#4383]
**Describe the bug**
What the bug is, and how to reproduce, better with screenshots(描述bug以及复现过程，最好有截图)

When using GRPO training in async mode, the model consistently generates identical completions for the same prompts, resulting in zero loss/KL divergence even after extended training (200+ steps). This behavior does not occur in colocated mode.

**observations:**
- Loss and KL divergence remain at zero throughout training
- Reward standard deviation is zero, while the reward function itself produces normal standard deviation
- Investigation reveals that identical prompts consistently produce identical completions
- Expected behavior: Different completions should be generated for the same prompt due to sampling randomness

#### 1. Start Rollout Node

```bash
CUDA_VISIBLE_DEVICES=0,1 \
swift rollout \
  --model Qwen/Qwen2.5-7B-Instruct \
  --tensor_parallel_size 2 \
  --data_parallel_size 1 \
  --max_model_len 16000 \
  --port 8000
```

#### 2. Start Training Node

```bash
CUDA_VISIBLE_DEVICES=0,1,2 \
NPROC_PER_NODE=3 \
swift rlhf \
    --rlhf_type grpo \
    --model Qwen/Qwen2.5-7B-Instruct \
    --reward_funcs accuracy \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host x.x.x.x \
    --vllm_server_port 8000 \
    --vllm_server_timeout 120 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset AI-MO/NuminaMath-TIR#1000 \
    --max_completion_length 8128 \
    --num_train_epochs 1 \
    --max_steps 500 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 4 \
    --learning_rate 5e-5 \
    --lr_scheduler_type cosine \
    --gradient_accumulation_steps 4 \
    --eval_steps 200 \
    --save_steps 100 \
    --save_total_limit 3 \
    --logging_steps 5 \
    --max_length 16000 \
    --output_dir output/example_train \
    --warmup_ratio 0.03 \
    --dataloader_num_workers 2 \
    --dataset_num_proc 2 \
    --num_generations 4 \
    --temperature 0.8 \
    --top_p 0.95 \
    --deepspeed zero3_offload \
    --log_completions true \
    --report_to wandb
```

### Evidence

WandB logs show duplicate entries with identical prompts producing identical completions. Despite using sampling parameters (`temperature=0.8`, `top_p=0.95`), the model generates deterministic outputs.

**Expected behavior**: Even with the same prompt, different completions should be generated due to the stochastic nature of sampling.

![Image](https://github.com/user-attachments/assets/60ff3224-d74b-4b1c-836e-7338ea45a353)

**Your hardware and system info**
Write your system info like CUDA version/system/GPU/torch version here(在这里给出硬件信息和系统信息，如CUDA版本，系统，GPU型号和torch版本等)
Rollout Node:
```
=== ms-swift ===
ms_swift @ git+https://github.com/modelscope/ms-swift.git@a0e18e9d92ae4516970bd94c19cc5991a0760fda
=== System Info ===
Ubuntu 22.04
Linux 6.8.0-59-generic #61~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Apr 15 17:03:15 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
=== CUDA Version ===
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2025 NVIDIA Corporation
Built on Wed_Apr__9_19:24:57_PDT_2025
Cuda compilation tools, release 12.9, V12.9.41
Build cuda_12.9.r12.9/compiler.35813241_0
=== GPU Info ===
NVIDIA GeForce RTX 4090, 565.57.01, 49140
NVIDIA GeForce RTX 4090, 565.57.01, 49140
=== Python/PyTorch Version ===
PyTorch: 2.6.0+cu124
CUDA Available: True
CUDA Version (PyTorch): 12.4
GPU Count: 2
```
Training Node:
```
=== ms-swift ===
ms_swift @ git+https://github.com/modelscope/ms-swift.git@6d00d463e3c816d22227a0935102f26588245376
=== System Info ===
Ubuntu 24.04
Linux 6.8.0-60-generic #63-Ubuntu SMP PREEMPT_DYNAMIC Tue Apr 15 19:04:15 UTC 2025 x86_64 x86_64 x86_64 GNU/Linux
=== CUDA Version ===
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2024 NVIDIA Corporation
Built on Tue_Oct_29_23:50:19_PDT_2024
Cuda compilation tools, release 12.6, V12.6.85
Build cuda_12.6.r12.6/compiler.35059454_0
=== GPU Info ===
NVIDIA GeForce RTX 3080, 570.133.20, 20480
NVIDIA GeForce RTX 4090, 570.133.20, 49140
NVIDIA GeForce RTX 3080, 570.133.20, 20480
=== Python/PyTorch Version ===
PyTorch: 2.6.0+cu124
CUDA Available: True
CUDA Version (PyTorch): 12.4
GPU Count: 3
```
**Additional context**
Add any other context about the problem here(在这里补充其他信息)
This issue specifically occurs with async/distributed training setup
Colocated mode works as expected. (But tested with lora training instead)
Initially tested with Qwen3 but switched to Qwen2.5-7B-Instruct for this reproduction case
